snapcraft/internal/parser.py

```diff
@@ -230,33 +233,37 @@ class Options:
 
 
 def _process_wiki_entry(entry, master_parts_list):
+    # return the number of errors encountered
     try:
         data = yaml.load(entry)
     except ScannerError as e:
         logger.warning(
             'Bad wiki entry, possibly malformed YAML: {}'.format(e))
-        return
+        return 1
 
     part_name = data.get('project-part')
     if part_name is not None and part_name in master_parts_list:
         logger.warning("Duplicate part found in wiki: {}".format(
             part_name))
-        return
+        return 1
 
     try:
         parts_list, after_parts = _process_entry(data)
     except InvalidEntryError as e:
         logger.warning('Invalid wiki entry: {}'.format(e))
-        return
+        return 1
 
     if is_valid_parts_list(parts_list, after_parts):
         master_parts_list.update(parts_list)
 
+    return 0
+
 
 def _process_index(output):
     # XXX: This can't remain in memory if the list gets very large, but it
     # should be okay for now.
     master_parts_list = OrderedDict()
+    errors = dict(wiki=0)
```

Comment:
The thinking is that we'd have at least two classes of errors; 1. wiki entry errors (i.e. malformed yaml, missing fields, etc.) 2. snapcraft.yaml errors that make constructing the fully parsed part impossible.  For either one we want to be able to continue processing the subsequent entries.  I'll look into handling this with exceptions.